src/loaders/ncbi_downloader/ncbi_downloader.py

```diff
@@ -1,15 +1,75 @@
 """
-Download genome files from NCBI FTP server
-"""
+PROTOTYPE - Download genome files from NCBI FTP server.
+
+usage: ncbi_downloader.py [-h] --download_file_ext
+                          DOWNLOAD_FILE_EXT
+                          [DOWNLOAD_FILE_EXT ...] --release_ver
+                          {207,202,95,89,86,83,80}
+                          [--root_dir ROOT_DIR]
+                          [--collection COLLECTION]
+                          [--threads THREADS]
+                          [--chuck_size CHUCK_SIZE] [--overwrite]
+
+options:
+  -h, --help            show this help message and exit
+
+required named arguments:
+  --download_file_ext DOWNLOAD_FILE_EXT [DOWNLOAD_FILE_EXT ...]
+                        Download only files that match given
+                        extensions.
+  --release_ver {207,202,95,89,86,83,80}
+                        GTDB release version
+
+optional arguments:
+  --root_dir ROOT_DIR   Root directory. (default: /global/cfs/cdi
+                        rs/kbase/collections/sourcedata)
+  --collection COLLECTION
+                        Collection (default: GTDB)
+  --threads THREADS     Number of threads. (default: half of
+                        system cpu count)
+  --chuck_size CHUCK_SIZE
+                        Number of genomes per thread
+  --overwrite           Overwrite existing files.
+
+
+e.g.
+python ncbi_downloader.py --download_file_ext genomic.gff.gz genomic.fna.gz --release_ver 207
+
+NOTE:
+NERSC file structure for GTDB:
+/global/cfs/cdirs/kbase/collections/sourcedata/ -> GTDB -> [GTDB_release_version] -> [genome_id] -> genome files
 
+e.g.
+/global/cfs/cdirs/kbase/collections/sourcedata/GTDB -> r207 -> GB_GCA_000016605.1 -> genome files
+                                                            -> GB_GCA_000200715.1 -> genome files
+                                                       r202 -> GB_GCA_000016605.1 -> genome files
+                                                            -> GB_GCA_000200715.1 -> genome files
+                                                       r80  -> GB_GCA_000016605.1 -> genome files
+                                                            -> GB_GCA_000200715.1 -> genome files
+
```

Comment:
Presumably at some point we'll want to update this script to softlink to earlier GTDB versions when the genome is the same

Reply:
👍